application-xpoll-api/src/main/java/com/xwiki/xpoll/internal/DefaultXPollManager.java

```diff
@@ -81,12 +86,16 @@
     @Named("compactwiki")
     private EntityReferenceSerializer<String> serializer;
 
+    @Inject
+    @Named("context")
+    private Provider<ComponentManager> componentManagerProvider;
+
     @Override
     public void vote(DocumentReference docReference, DocumentReference user, List<String> votedProposals)
         throws XPollException
     {
         XWikiContext context = contextProvider.get();
-        XWikiDocument document = null;
+        XWikiDocument document;
         try {
             document = context.getWiki().getDocument(docReference, context);
```

Comment:
```suggestion
            document = context.getWiki().getDocument(docReference, context).clone();
```

We should clone the document here, before modifying it in ``setUserVotes`` and ``updateWinner``, otherwise we risk modifying the cached document which might lead to unexpected side effects.